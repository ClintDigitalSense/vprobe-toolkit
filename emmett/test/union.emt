typedef union {
   uint8  a;
   uint32 b;
   uint64 c;
} u;

typedef struct {
   u u1;
   u u2;
} s;

s* sptr;
VMM1Hz printf("%d %d %d\n", &sptr->u1, &sptr->u2, &sptr->u2.c);
