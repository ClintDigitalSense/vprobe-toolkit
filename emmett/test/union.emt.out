(version 0.2)
(definteger sptr)

 ;;;;;
(vprobe VMM1Hz ( printf    "%d %d %d\n"  sptr   (+ sptr  8 )   (+ sptr  8 ) ))